proxy/src/coco/peer.rs

```diff
@@ -603,6 +665,72 @@ mod test {
         Ok(())
     }
 
+    #[tokio::test]
+    async fn test_can_clone_project() -> Result<(), Error> {
+        pretty_env_logger::init();
+
+        let mut bob_addr = *config::LOCALHOST_ANY;
+        bob_addr.set_port(8066);
+
+        let mut alice_addr = *config::LOCALHOST_ANY;
+        alice_addr.set_port(8067);
+
+        let alice_key = SecretKey::new();
+
+        let alice_tmp_dir = tempfile::tempdir().expect("failed to create tempdir");
+        let alice_repo_path = alice_tmp_dir.path().join("radicle");
+        let paths = paths::Paths::from_root(alice_tmp_dir.path())?;
+
+        let config = config::configure(paths, alice_key.clone(), alice_addr, vec![]);
+        let alice_peer = Api::new(config).await?;
+
+        let alice = alice_peer.init_owner(alice_key.clone(), "alice")?;
+        let project =
+            alice_peer.init_project(&alice_key, &alice, &alice_repo_path, "just", "do", "it")?;
+
+        let bob_key = SecretKey::new();
+
+        let path = {
+            let tmp_dir = tempfile::tempdir().expect("failed to create tempdir");
+            let path = tmp_dir.into_path();
+            std::fs::remove_dir_all(path.clone()).expect("failed to remove tempdir");
+            path
+        };
+
+        std::fs::create_dir(path.clone()).expect("failed to create directory");
+
+        // let bob_tmp_dir = tempfile::tempdir().expect("failed to create tempdir");
+        let bob_paths = paths::Paths::from_root(path)?;
+        let bob_config = config::configure(bob_paths, bob_key.clone(), bob_addr, vec![]);
+        let bob_peer = Api::new(bob_config).await?;
+        let _bob = bob_peer.init_owner(bob_key, "bob")?;
+
+        let bobby = bob_peer.clone();
+
+        // TODO(sos): move spawn_blocking into clone_project & clone_user
```

Comment:
I tried this and it doesn't play nicely. We'll just have to spawn_blocking wherever we call these. Can you add that to the docs of the functions please?

Reply:
Sounds good!